src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -251,13 +251,22 @@ public override FormulaValue Visit(BinaryOpNode node, SymbolContext context)
                         {
                             if (cov.Impl.TryGetProperty(sv.Value, out var res))
                             {
+                                if (res.IsNull)
+                                {
+                                    return new BlankValue(node.IRContext);
+                                }
+
                                 return new CustomObjectValue(node.IRContext, res);
                             }
                             else
                             {
                                 return new BlankValue(node.IRContext);
                             }
                         }
+                        else if (cov.Impl.IsNull)
+                        {
```

Comment:
this check here is what prompted me wondering about being an enum instead of multiple IsXXX functions. 

Because if an impl did haev IsNull and IsObj both true, then the order of checks here would start to matter.

Reply:
Changed to an enum